core/src/test/java/kafka/server/SharePartitionManagerTest.java

```diff
@@ -1107,6 +1110,50 @@ public void testAcknowledgeSuccess() {
         assertEquals(Errors.INVALID_REQUEST.code(), result.get(tp2).errorCode());
         assertEquals(4, result.get(tp3).partitionIndex());
         assertEquals(Errors.UNKNOWN_TOPIC_OR_PARTITION.code(), result.get(tp3).errorCode());
+
+        Map<MetricName, Consumer<Double>> expectedMetrics = new HashMap<>();
+        expectedMetrics.put(
+            metrics.metricName(SharePartitionManager.ShareGroupMetrics.SHARE_ACK_COUNT, SharePartitionManager.ShareGroupMetrics.METRICS_GROUP_NAME),
+            val -> assertEquals(val, 1.0)
```

Comment:
Shouldn't it record the number of offsets acknowledged, will be more than 1?

Reply:
As per the definition provided by @AndrewJSchofield 
```
the first one is the number of requests to acknowledge
```